autogen/agentchat/contrib/vectordb/pgvectordb.py

```diff
@@ -605,16 +612,21 @@ def __init__(
                     f"@{encoded_host}:{parsed_connection.port}/{encoded_database}"
                 )
                 self.client = psycopg.connect(conninfo=connection_string_encoded, autocommit=True)
-            elif host and port and dbname:
+            elif host and port and dbname and username and password:
```

Comment:
Are all these required to connect to a postgres db? Is it possible to do something like the following:
```python
connection_string = ""
if host and port:
    encoded_host = urllib.parse.quote(host, safe="")
    connection_string += f"host={encoded_host} port={port}" 
if dbname:
    encoded_database = urllib.parse.quote(dbname, safe="")
    connection_string += f" dbname={encoded_database}" 
```
etc... This logic should also be in its own function to keep `__init__` clean

Reply:
That is a good question, I assumed the database name was always required. Let me validate.

Reply:
I have updated to build the connection string as specified in a separate function outside of init.